index.rst

```diff
@@ -505,15 +505,117 @@ on the producer end.
 Conclusions
 -----------
 
-TO BE EXTENDED AND ORGANIZED IN TICKET DM-13307.
-
-For most experiments and metrics measured, assuming that the size of alerts has been estimated accurately,
-there is a significant impact on Kafka compute utilization and
-timing when postage stamp cutouts are included in the alert packets.
-The inclusion of postage stamp cutouts is currently a science requirement.  However,
-results here suggest that additional consideration be
-given to alternatives for the distribution of stamps, if it is possible
-that the current requirements may be adjusted.  One alternative to including postage
+Compute Resources and Settings Recommendations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+For a robust alert distribution service, a cluster of at least three Kafka brokers and a replication factor of two
+is recommended.  However, we did not test here the performance effects of a replication factor greater than one.
+
+Kafka is a memory intensive application.  In the experiments here, each Kafka broker tends to utilize all
+available memory, which it needs to buffer active producers and consumers.
+In terms of hardware recommendations, Confluent, the main contributor to Kafka, recommends a machine with
+64 GB of RAM or an AWS instances with 32 GB.  An appropriate back of the envelope calculation to determine
+hardware needs is `write_throughput`*30 if you want a 30 second buffer.  If we assume one alert is ~135 KB
+and possible bursts of up to 10,000 alerts/second, that is equivalent to 1.35 GB/second.  If we want to buffer
+for between 30-60 seconds, given the time between visits, an appropriate memory range is then between ~40-80 GB.
+
+In all experiments above, Kafka tends to be low on CPU requirements.  It is noted by Confluent that enabling
+SSL can significantly increase the CPU requirements.  A reasonable recommendation is for 24 core machines.
+
+The requirements for storage disk sizes depend on the amount of time we will cache / allow "rewindable" access
+to the alert stream and the number of partitions configured for each topic.  If all alerts are streamed out
+in a single topic and the alert sizing assumptions here are correct, one night of minimal alerts is
+about ~1.5 TB.  Each partition of a topic must fit on its own disk, and it is recommended that these disks
```

Comment:
Does “disk“ really mean “disk” here?

I ask in a couple of senses:

- Does it actually mean separate media, or just different (storage, not Kafka) partitions? Below you talk about needing 128 x 12 GB disks for example; that sounds like some kind of weird early 2000s massive RAID system. Presumably 128 (storage) partitions on a small number of physical disks would do?
- Can this be stored on e.g. GPFS, or does it really need physically attached hardware?

Reply:
Yes, sorry, you don't need to split partitions to their own disk, you need a disk of at least some minimum size to fit a whole (or multiple) full partitions because you can't split partitions over multiple disks.  I fixed this to clarify.  Also, I can't find anyone that says they use GPFS, but I found some remarks that any POSIX compliant system is ok.